### DISCUSSION TITLE
[Suggestion] Add Instant Search to the core?

### DISCUSSION THREAD

**Discussion by scottcwilson** [zencart/zencart#5771] [Ideas]
Link: 
https://www.zen-cart.com/downloads.php?do=file&id=1336

**Comment by torvista** [comment]
Yes for me, been using a customised version for years.

**Comment by drbyte** [comment]
@torvista in what ways did you customize it? Any of those worth incorporating, or are they very specialized?

**Comment by torvista** [comment]
Bizarrely I was forced to start looking at this today for the first time in years due to an issue with it today on my dev shop.
So, I've only just started to compare it with previous/current Plugin versions. So, not sure what my all changes are/were for yet, apart from returning Product Options Stock-based product results....

**Comment by torvista** [comment]
For those itching to give this the twice-over to make it more ZC-like, there is a now a repository.
https://github.com/davesaddle/InstantSearch

**Comment by ghost** [comment]
I'll give it a try.. I'll have to look at the documentation for PR/development I was about to upgrade to PHP7.3.8....  Is there a wish list of what needs to be auto searched for?

**Comment by Zen4All** [comment]
@davesaddle I know you volunteered to do this ,but I have an updated version currently sitting in my [repo ](https://github.com/Zen4All-nl/InstantSearch/releases/tag/v2.0.0-beta.1)
It needs some refinements, but can be used, and used as a template for files used in a PR.

@torvista: Done
> For those itching to give this the twice-over to make it more ZC-like, there is a now a repository.
> https://github.com/davesaddle/InstantSearch

Any input from both of you, or anyone else is more than welcome

**Comment by torvista** [comment]
Ok, I'll try and compare in the next week...

**Comment by torvista** [comment]
I'm fighting on too many fronts to look at this so I have just put up my version, it should be commented enough to figure out what I changed.
https://github.com/torvista/instant-search-steve

**Comment by Zen4All** [comment]
> many fronts to look at this so I

Thanks Steve, No problem, I will figure it out.
I'll probably also release this as a new version of the module

**Edit:**

I had a quick look, and need some time to review the differences, I'll probably  will have time to do this next week

**Comment by lat9** [comment]
I'll note that this feature has been added to the [Bootstrap Template](https://github.com/lat9/ZCA-Bootstrap-Template/issues/99).  That could be used as a model for the base Zen Cart update.

**Comment by torvista** [comment]
IMHO this is the first feature that should be included in core once the two versions are reviewed/consolidated.

**Comment by zcwilt** [comment]
Is this still using FULLTEXT search. If so not a fan.

**Reply by lat9** [reply]
The 'Instant Search' (https://www.zen-cart.com/downloads.php?do=file&id=1336) plugin does; the Bootstrap template one doesn't (but it's also not as flexible).

**Comment by marco-pm** [comment]
I know it may seem strange that I advocate "against" using the Instant Search plugin, but all in all I think the best option would be to include the Bootstrap template in the core and consequently also its Ajax search function.
Easier to maintain, similar logic to storefront search, and (from what I've experienced) quite fast in showing results.
Then, if a user prefers to have more flexibility, then they can consider installing the Instant Search plugin.
Just my 2 cents.

**Comment by torvista** [comment]
> Just my 2 cents.

**Your** comments **here** are a great deal more valuable than that....it's hard to recommend retirement for your own work, especially as it is an exemplary plugin.

**Comment by marco-pm** [comment]
> it's hard to recommend retirement for your own work, especially as it is an exemplary plugin.

Never talked about retirement in fact! As I said, it would remain an optional plugin, as it is now.
Most important thing imho is to include the boostrap template in the core. This way you will also get Ajax search as a bonus.

**Reply by webconcept-one** [reply]
Nothing bad about Bootstrap but I wrote some basic framework myself for use that on a Front-Page with responsive_classic.
So no need for a big framework like Bootstrap as long you don't use Bootstrap e.g. with the Bootstrap theme.

Instant Search is surprisingly ressource friendly and a real benefit to the user.
Installation of Instant Search is so clean and simple that no need to bloat the Core IMHO.

**Comment by torvista** [comment]
>Most important thing imho is to include the bootstrap template in the core.

There's a whole other discussion needed for that....three-in-built templates to maintain...something has to go!
And the Bootstrap template still seems to be an actively-moving target.

**Reply by dbltoe** [reply]
I would think dropping both classic and default. Neither is helpful anymore IMHO. Having responsive as the default at least gives the store owner an operational cart out-of-the-box. 
and, while we’re at it, get Clone a Template in the core with a note in Template Selection to clone ASAP

**Reply by webconcept-one** [reply]
If it is work to maintain classic and default I also would think to drop them both and take one usable and editable starting point with responsive_classic.

**Comment by lat9** [comment]
Agreed, while it would be more convenient for installing sites if the Bootstrap Template was in-core, per @torvista's comment, the template has its own release cycle and tying it to the core would make maintenance more of a pain for both.

**Comment by drbyte** [comment]
regardless of whether the front-end javascript is triggered by bootstrap or some other library or even some isolated js code, the back-end logic can probably be standardized and built-in to core.

Which of the available implementations/plugins/contributions would make the most sense to be incorporated to core? Are there other features from other ones that can be added on as well? Again, I'm talking about non-template logic here.

**Comment by scottcwilson** [comment]
I would vote to close this discussion, which is 3.5 years old, since Bootstrap already builds in a nice search, and the plugin can easily be added to other templates.
